### PR TITLE
Support for spaces in package variable declarations

### PR DESCRIPTION
**Pull request by shakerlxxv**
Fixes rsim/ruby-plsql#111

### CODE REVIEW

File: spec/plsql/variable_spec.rb [hunk 1]
```diff
@@ -99,6 +102,12 @@
       expect(plsql.test_package.nchar_variable).to eq('abc' + ' '*7)
     end
 
+    it 'can use package variables with spaces in size' do
+      expect(plsql.test_package.varchar_with_spaces1).to eq('a string')
+      expect(plsql.test_package.varchar_with_spaces2).to eq('a string')
```
Comment (javornikolov) [inline spec/plsql/variable_spec.rb]: Multiple expectations in same example is in general a bit problematic (if 1st fails one fails, the rest won't b executed). I'm wondering if we can apply either RSpec [aggregating_faulures](https://relishapp.com/rspec/rspec-core/docs/expectation-framework-integration/aggregating-failures) or some of the ways for parameterised examples

**Comment by javornikolov** [comment]
@shakerlxxv, thanks a lot for the pull request. In general looks good - just I posted one comment about the test which is good to consider.

File: spec/plsql/variable_spec.rb (hunk 1 above)
Comment (shakerlxxv) [inline spec/plsql/variable_spec.rb]: Good suggestion on the `aggregating_failures`. I'll make that update.

File: lib/plsql/variable.rb [hunk 2]
```diff
@@ -47,7 +47,7 @@ def value=(new_value)
 
     def metadata(type_string)
       case type_string
-      when /^(VARCHAR|VARCHAR2|CHAR|NVARCHAR2|NCHAR)(\((\d+)[\s\w]*\))?$/
+      when /^(VARCHAR|VARCHAR2|CHAR|NVARCHAR2|NCHAR)(?:\s+)?(\((?:\s+)?(\d+)[\s\w]*\))?$/
```
Comment (javornikolov) [inline lib/plsql/variable.rb]: I'm wondering if this can be simplified a bit. E.g. what about using `\s*` instead of `(\s+)?`?

Comment (shakerlxxv) [inline lib/plsql/variable.rb]: Yes, it looks like you're correct.

**Pull request closed without merging**